File: patterns/encoding-and-accessing/rules-python/encoding_and_accessing.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger

PRESENT_TARGETS = {
    "Room1_Thermostat": 21,
    "Room2_Thermostat": 22,
}
AWAY_TARGETS = {
    "Room1_Thermostat": 17,
    "Room2_Thermostat": 18,
}


@rule(triggers=[ItemStateChangeTrigger("vPresent")])
class ZieltemperaturenAnwenden:
    def execute(self, module, input):
        targets = PRESENT_TARGETS if str(Registry.getItem("vPresent").state) == "ON" else AWAY_TARGETS
        for name, temp in targets.items():
            Registry.getItem(name).sendCommand(temp)

File: patterns/encoding-and-accessing/rules-js/encoding-and-accessing.js
const presentTargets = {
  "Room1_Thermostat": 21,
  "Room2_Thermostat": 22
};
const awayTargets = {
  "Room1_Thermostat": 17,
  "Room2_Thermostat": 18
};

rules.JSRule({
  name: "Zieltemperaturen anwenden",
  triggers: [triggers.ItemStateChangeTrigger("vPresent")],
  execute: (event) => {
    const targets = items.getItem("vPresent").state === "ON" ? presentTargets : awayTargets;
    Object.entries(targets).forEach(([name, temp]) => {
      items.getItem(name).sendCommand(temp);
    });
  }
});
